config/config.go

```diff
@@ -44,6 +45,7 @@ type Config struct {
 	Identity               security.Identity ",omitempty"
 	Server                 Server            ",omitempty"
 	Prometheus_Service_Url string            ",omitempty"
+	Istio_Identity_Domain  string            ",omitempty"
 }
 
```

Comment:
trivial: I'm no expert but I'm fairly sure the Go convention for naming encourages MixedCase as opposed to all caps or underscores.  Perhaps we should be using names like:

EnvServerAddress
PrometheusServiceUrl

I'm not sure, just looking at some other examples and https://golang.org/doc/effective_go.html#names

Reply:
This is due the yaml parser, to read prometheus-service-url.

Reply:
This is why there is a all-caps comment on this object about "USED FOR YAML"

Because the Yaml parser cannot be told in the ",omitempty" section what the yaml property is to be called, it is forced to determine this from the variable name. And how it does this is simply make the variable name all lowercase with no demarcation characters like - or _.

So if we do not name them with underscores, the words in the property names will all run together --- if we did it with your suggestion, for example, the yaml file would have properties called "istioidentitydomain" and "prometheusserviceurl". Difficult to read. So we use underscores between words so when you write the yaml file the properties are "istio_identity_domain" and "prometheus_service_url" - much easier to read and understand.

So in short the variable names in the structs that are used for YAML processing have _ between words. Look for the comment at the top of structs - if you see USED FOR YAML then we will use this naming convention. Otherwise, we do use camel case.

Reply:
OK, thanks. I think actually we talked about this off-pr but I forgot.  The PR diff didn't show that comment so I didn't see it (although the comment doesn't mention that this unconventional format is needed).  Anyway, no problem, carry on.

Reply:
Unless I'm missed something obvious, you could use struct tags to stick with the Golang conventions:

```
type Config struct {
	Identity             security.Identity ",omitempty"
	Server               Server            ",omitempty"
	PrometheusServiceUrl string            "prometheus_service_url,omitempty"
	IstioIdentityDomain  string            "istio_identity_domain,omitempty"
}
```

It's widely used in the Prometheus code base, see https://github.com/prometheus/prometheus/blob/master/config/config.go#L267-L279 for instance.

Reply:
AFAIK, this was not supported in the version of yaml parser at the time I originally started using this yaml parser. If this is now a supported feature of the yaml parser, then yes, this is what we should do.

Reply:
https://godoc.org/gopkg.in/yaml.v2

Yes, looks supported. Feel free to refactor our config structures with this.

Reply:
If no one is doing it already, I can refactor the config structures.

Reply:
@jshaughn vscode shows me warning about it, but I preferred to stick with the rules adopted with the other settings.

In a similar way it is also showing warning for all the const defined above (like ENV_IDENTITY_CERT_FILE) : "don't use ALL_CAPS in Go names; use CamelCase"

Reply:
I sent a PR about the config structures: https://github.com/swift-sunshine/swscore/pull/33